### DISCUSSION TITLE
"canvas.scale" call in "sheet_music_renderer.dart"

### DISCUSSION THREAD

**Discussion by lukasnalezenec** [tomoyu719/simple_sheet_music#24] [Q&A] [answered]
Hi,
What is the function of this  canvas.scale call?
`  

void paint(Canvas canvas, Size size) {
    canvas.scale(sheetMusicLayout.canvasScale);
    sheetMusicLayout.render(canvas, size);
  }

`

https://github.com/tomoyu719/simple_sheet_music/blob/69571c2fd23fd6b55c0c2869c18848f688b02669/lib/src/sheet_music_renderer.dart#L12C4-L12C18

**Comment by tomoyu719** [comment, marked as answer]
@lukasnalezenec 
Thank you for your question. [canvas.scale](https://github.com/tomoyu719/simple_sheet_music/blob/69571c2fd23fd6b55c0c2869c18848f688b02669/lib/src/sheet_music_renderer.dart#L12C4-L12C18) is required to adjust [SimpleSheetMusicWidget](https://github.com/tomoyu719/simple_sheet_music/blob/69571c2fd23fd6b55c0c2869c18848f688b02669/lib/src/sheet_music_renderer.dart#L5) to the specified [width](https://github.com/tomoyu719/simple_sheet_music/blob/69571c2fd23fd6b55c0c2869c18848f688b02669/lib/src/simple_sheet_music.dart#L33) and [height](https://github.com/tomoyu719/simple_sheet_music/blob/69571c2fd23fd6b55c0c2869c18848f688b02669/lib/src/simple_sheet_music.dart#L32). The library first draws the music symbol in full size to Canvas, calculates the scale based on the specified width and height from that size, and then applies it to Canvas, usually scaled down. (Try commenting out canvas.scale and you should see a huge music symbol).

**Comment by lukasnalezenec** [comment]
"Thank you very much for the clarification. It's interesting; I would not have expected that."
